hudi-common/src/main/java/org/apache/hudi/common/model/HoodieReplaceCommitMetadata.java

```diff
@@ -115,12 +117,32 @@ public int hashCode() {
     return result;
   }
 
-  public static <T> T fromBytes(byte[] bytes, Class<T> clazz) throws IOException {
+  /*public static <T> T fromBytes(byte[] bytes, Class<T> clazz) throws IOException {
     try {
       return fromJsonString(fromUTF8Bytes(bytes), clazz);
     } catch (Exception e) {
       throw new IOException("unable to read commit metadata", e);
     }
+  }*/
+
+  public static <T> T fromBytes(byte[] bytes, Class<T> clazz) throws IOException {
+    try {
+      if (bytes.length == 0) {
+        return clazz.newInstance();
+      }
+      return fromJsonString(
+          fromUTF8Bytes(
+              convertCommitMetadataToJsonBytes(deserializeReplaceCommitMetadata(bytes), org.apache.hudi.avro.model.HoodieReplaceCommitMetadata.class)),
+          clazz);
+    } catch (Exception e) {
+      LOG.info("Failed to deserialize replace commit metadata as avro, trying json ");
+      try {
```

Comment:
NTR: same as above. deser as avro. on exception we can do json. 
I have not introduced the table upgrade commit time change as proposed in the rfc. once we introduce that, we should be able to avoid this.